### DISCUSSION TITLE
[Feature Request] Ability to select Type of review upon submission

### DISCUSSION THREAD

**Discussion by James-Firth** [ReviewNB/support#114] [Ideas]
When reviewing a PR on Github I have the ability add a comment about the review overall, and can specify whether it's a comment, approval, or requesting changes.

We're currently trying out ReviewNB and we require 2 Approvals to merge a PR on Github.

**Problem**
The Problem I'm encountering is ReviewNB only submits a "comment" type review. I generally want to request changes, which will block the PR from being merged even if there are other approvals.

To handle this now I have to manually go to the PR on github and submit a second review to set that I have requested changes.

**Requested Feature**

I would like to be able to specify the type of review to post when choosing "Finish Review"

Screenshot from github review for reference:
![image](https://user-images.githubusercontent.com/485809/181586380-4e39fe07-33dd-42a3-a21f-a795af895af0.png)

**Comment by amit1rrr** [comment]
Thank you @James-Firth for starting this conversation & explaining your use case.

We had this exact feature request https://github.com/ReviewNB/support/issues/78 open for a while in the past & received little interest.

> I generally want to request changes, which will block the PR from being merged even if there are other approvals.

From what I understand, "Request Changes" does not block merge but simply shows that changes are requested right above the Merge button. Is that correct? You likely have a (reasonable) team policy to not merge when there are pending requested changes, right?

**Reply by James-Firth** [reply]
Ah, yes you're correct!

All the projects I've been on at work have a policy, enforced by a branch protection rule on github.

Here's the setting we're currently using changes must be in a PR to get to our `main` branch _and_ it must have 2 approvals
![image](https://user-images.githubusercontent.com/485809/181796316-888b3724-11cf-4e9f-9555-f947df25e213.png)



I'm a bit surprised #78 didn't get more traction! I'd love to see it as an option, even if it just defaults to a "Comment" and there's an extra step to pick the Approve or Request changes.

**Reply by amit1rrr** [reply]
Thanks for the info. Since "Request Changes" does not actually block the merge but just informs the author - Is it possible for you to consider team policy along the lines of - '_PR will not be merged until all conversations are resolved_'. Most teams we talk to seem to follow something along these lines. This will be in addition to your 2-approval mechanism.

Given "Request Changes" has little impact on the workflow (one can still merge) & little interest shown by our users previously, it's likely not going to be on our roadmap soon. Just want to give you a fair idea.

**Reply by James-Firth** [reply]
Apologies for the late reply, I missed this notification

---
Oh interesting, I didn't notice that in the branch protection options previously.

I'll have our team try out that policy and see how it works out!

Appreciate the transparency and suggestions, thank you!
